From cbe714fc2a3b57526cf9d930406dc78baa35c5d9 Mon Sep 17 00:00:00 2001
From: intrigeri <intrigeri@boum.org>
Date: Thu, 7 Dec 2017 05:54:27 +0000
Subject: [PATCH] Nautilus extension: avoid spamming the logs due to missing method.

Without this change I see a dozen or so warnings in the Journal:

  nautilus_menu_provider_get_background_items: assertion
  'NAUTILUS_IS_MENU_PROVIDER (provider)' failed

Nautilus upstream developers are going to discuss if/how they want to address
this on their side; in the meantime they recommend simply adding a no-op method
to silence the logs, so here we go.
---
 nautilus/nautilus-mat.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nautilus/nautilus-mat.py b/nautilus/nautilus-mat.py
index 7c2d740..ff102b8 100644
--- a/nautilus/nautilus-mat.py
+++ b/nautilus/nautilus-mat.py
@@ -93,3 +93,7 @@ class MatExtension(GObject.GObject, Nautilus.MenuProvider):
                 self.show_message(_("Unable to clean %s") % file_path, Gtk.MessageType.ERROR)
         else:
             self.show_message(_("Unable to process %s") % file_path, Gtk.MessageType.ERROR)
+
+    # Workaround https://bugzilla.gnome.org/show_bug.cgi?id=784278
+    def get_background_items(self, window, file):
+        return None
--
libgit2 0.26.0

